schematic/utils/schema_utils.py

```diff
@@ -207,15 +214,156 @@ def convert_bool_to_str(provided_bool: bool) -> str:
     return str(provided_bool)
 
 
-def parse_validation_rules(validation_rules: List[str]) -> List[str]:
+def get_individual_rules(rule: str, validation_rules: list[str]) -> list:
+    # Separate multiple rules (defined by addition of the rule delimiter)
+    if DELIMITERS["rule_delimiter"] in rule:
+        validation_rules.append(parse_single_set_validation_rules(rule))
+    # Get single rule
+    else:
+        validation_rules.append(rule)
+    return validation_rules
+
+
+def get_component_name_rules(
+    component_names: list, component_rule: str
+) -> Tuple[list, str]:
+    # If a component name is not attached to the rule, have it apply to all other components
+    if DELIMITERS["component_name_delimiter"] != component_rule[0]:
+        component_names.append("all_other_components")
+    # Get the component name if available
+    else:
+        component_names.append(
+            component_rule.split(" ")[0].replace(
+                DELIMITERS["component_name_delimiter"], ""
+            )
+        )
+        try:
+            assert component_names[-1] != " "
+        except ValueError:
+            print(
```

Comment:
This try/except block is a little wierd.. If you want to print that message when
`component_names[-1] != " "` then why not an `if` statement?
```
if component_names[-1] != " ":
    print(msg)
```

Reply:
Well this is raising an Error if the assert doesn't work...I was following the pattern I had seen elsewhere in our code base but I could do an `if/else` and use `logger.error` instead like:
```
if component_names and component_names[-1] == " ":
    logger.error(" ...")
 ```

Reply:
Yeah I've seen it elsewhere in the codebase... I'm not sure where it came from :)

Reply:
Looking up docs, try:excepts seem standard...what is the method you would prefer?
https://docs.python.org/3/tutorial/errors.html

Reply:
They both likely do the same thing. To me it's more about writing clear and readable code  and this is a case of control flow: If <something is the case> : print (msg).

Using the try/except tells me that you are expecting unavoidable exceptions somewhere else and you need to catch them so you cna handle them. By adding the assert staement the code is forcing an unesseccary exception just so that a try/except statement block can be used.

I don't think it's a big deal, I when I saw it, it took me a moment to understand why it was written this way.
I think this is much clearer:

```
if component_names and component_names[-1] == " ":
    logger.error(" ...")
```